backend/api/Controllers/RobotController.cs

```diff
@@ -637,5 +639,86 @@ public async Task<ActionResult> ResumeMission([FromRoute] string robotId)
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> ResetRobot(
+            [FromRoute] string robotId
+        )
+        {
+            Console.WriteLine(robotId);
+            Console.WriteLine("oooooooooo....................");
+            var robot = await robotService.ReadById(robotId);
+            if (robot == null)
+            {
+                string errorMessage = $"Could not find robot with id {robotId}";
+                logger.LogWarning("{Message}", errorMessage);
+                return NotFound(errorMessage);
+            }
+
+            try { await missionSchedulingService.CancelAllScheduledMissions(robot.Id); }
+            catch (RobotNotFoundException)
+            {
+                string errorMessage = $"Failed to cancel scheduled missions for robot with id {robotId}";
+                logger.LogWarning("{Message}", errorMessage);
+                return NotFound(errorMessage);
+            }
+
+            try { await missionSchedulingService.StopCurrentMissionRun(robot.Id); }
+            catch (RobotNotFoundException)
+            {
+                string errorMessage = $"Failed to stop current mission for robot with id {robotId} because the robot was not found";
+                logger.LogWarning("{Message}", errorMessage);
+                return NotFound(errorMessage);
+            }
+            catch (MissionRunNotFoundException)
+            {
+                string errorMessage = $"Failed to stop current mission for robot with id {robotId} because the mission was not found";
+                logger.LogWarning("{Message}", errorMessage);
+                return Conflict(errorMessage);
+            }
+            catch (MissionException ex)
+            {
+                if (ex.IsarStatusCode != StatusCodes.Status409Conflict)
+                {
+                    string errorMessage = "Error while stopping ISAR mission";
+                    logger.LogError(ex, "{Message}", errorMessage);
+                    return Conflict(errorMessage);
+                }
+            }
+            catch (Exception ex)
+            {
+                string errorMessage = "Error in ISAR while stopping current mission";
+                logger.LogError(ex, "{Message}", errorMessage);
+                return Conflict(errorMessage);
+            }
+
+            try { await robotService.UpdateCurrentArea(robot.Id, null); }
+            catch (RobotNotFoundException)
+            {
+                string errorMessage = $"Failed to set current area to null for robot with id {robotId} because the robot was not found";
+                logger.LogWarning("{Message}", errorMessage);
+                return NotFound(errorMessage);
+            }
+
+            missionSchedulingService.TriggerRobotAvailable(new RobotAvailableEventArgs(robot.Id));
```

Comment:
Doesn't this happen by itself when all missions are cancelled and status becomes available? Do we risk it running twice now?

Reply:
I am not sure, can have a check on this later today

Reply:
You were right so I removed it